### DISCUSSION TITLE
Resize of manually created VM (macOS)

### DISCUSSION THREAD

**Discussion by dockay** [cirruslabs/tart#573] [Q&A] [answered]
First, thanks for the great tool! 

We are using it with manually creating VMs with `tart create --from-ipsw=latest ventura-vanilla --disk-size 60`

Now we ran into the issue, that the disk size we specified is not enough. Is there any way to extend the existing VM disk-size? Instead of re-creating everything from scratch with a bigger disk-size.

**Comment by winkelsdorf** [comment]
I found this related comment: https://github.com/cirruslabs/tart/issues/17#issuecomment-1439367869, but I would love to be able to have tart to automate this process if possible, too.

**Reply by dockay** [reply]
This does not work, if you have a recovery-partition installed. As we do have. I can't remove it, as I need to boot into recovery-mode, but there is no way to boot into it via command line anymore on ventura.

**Comment by dockay** [comment, marked as answer]
We now found a solution for our problem.

First there is a --recovery option for `tart run`. So you can run into recovery mode via `tart run vmname --recovery`.

On the host you then can use `truncate` to resize the disk.img to desired size.

Then start the vm with above command `tart run vmname --recovery`.

We had the issue, that we had an additional recovery partition between our free space and our container.

In recovery mode, disable csrutil `csrutil disable`.

`diskutil list` to find the Apple_APFS_recovery partition.

`diskutil apfs deletecontainer diskxx`

`diskutil repairdisk disk2`

You should now have the free space to your demand.

Find your apfs container you want to grow.

`diskutil apfs resizecontainer diskxx 0`

reenable csr.

`csrutil enable`

and reboot without --recovery.

**Reply by qorzi** [reply]
When executing the following command:
`diskutil apfs deletecontainer diskxx (Apple_APFS_recovery partition)`
it results in the deletion of disk2, which makes diskutil repairdisk disk2 impossible. What should I do?

i use :
```
tart clone ghcr.io/cirruslabs/macos-monterey-xcode:13.4 monterey
tart set monterey --disk-size 150
tart run monterey
```

**Reply by ethaniel86** [reply]
> When executing the following command: `diskutil apfs deletecontainer diskxx (Apple_APFS_recovery partition)` it results in the deletion of disk2, which makes diskutil repairdisk disk2 impossible. What should I do?
> 
> i use :
> 
> ```
> tart clone ghcr.io/cirruslabs/macos-monterey-xcode:13.4 monterey
> tart set monterey --disk-size 150
> tart run monterey
> ```

same here. Still have no idea how to resize newly created MacOS vm.

**Comment by fkorotkov** [comment]
Alternatively, one can use [Tart Packer Plugin](https://github.com/cirruslabs/packer-plugin-tart) to automate image creation. The plugin removes a recovery partition by default and can resize disks too. Here is a repository with [our example templates](https://github.com/cirruslabs/macos-image-templates).
